snap/gadget.go

```diff
@@ -69,6 +72,71 @@ type VolumeContent struct {
 	Unpack bool `yaml:"unpack"`
 }
 
+// GadgetConnect describes an interface connection requested by the gadget
+// between seeded snaps. The syntax looks like this:
+//
+// [<plug-snap-id1>|system]:plug [[<slot-snap-id>|system]:slot]
+//
+// "system" or omitting the snap-id indicates a system plug or slot.
+// Fully omitting the slot part indicates a system slot with the same name
+// as the plug.
+type GadgetConnect struct {
+	PlugSnapID string
+	Plug       string
+	SlotSnapID string
+	Slot       string
+}
+
+func parseSnapIDColonName(s string) (snapID, name string, err error) {
+	parts := strings.Split(s, ":")
+	if len(parts) == 2 {
+		snapID = parts[0]
+		name = parts[1]
+	}
+	if name == "" {
+		return "", "", fmt.Errorf("in gadget connect expected [snap-id]:name not %q", s)
+	}
+	if snapID == "system" {
+		snapID = ""
+	}
+	return snapID, name, nil
+}
+
+func (gconn *GadgetConnect) UnmarshalYAML(unmarshal func(interface{}) error) error {
```

Comment:
This is all fine but I think we need some new code in validate.go as well. Right now one could put any kind of garbage in snapID and that, at least, could be validated. We should also validate that interface names are using the right set of characters

Reply:
It wouldn't hurt to have basic validation there indeed.